be/src/formats/parquet/file_reader.cpp

```diff
@@ -489,16 +489,15 @@ void FileReader::_prepare_read_columns() {
 
 bool FileReader::_select_row_group(const tparquet::RowGroup& row_group) {
     size_t row_group_start = _get_row_group_start_offset(row_group);
-
-    for (auto* scan_range : _scanner_ctx->scan_ranges) {
+    const auto* scan_range = _scanner_ctx->scan_range;
+    if (scan_range != nullptr) {
```

Comment:
I think we don't need to check nullptr anymore.

Reply:
fixed